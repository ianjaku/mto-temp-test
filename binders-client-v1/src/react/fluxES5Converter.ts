/* eslint-disable @typescript-eslint/no-explicit-any */
// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export function fixES5FluxContainer<T extends new (...args: any[]) => any>(
    containerClass: T
): T {
    const tmp = containerClass;
    const newClass = function(...args: any[]) {
        return new tmp(...args);
    } as unknown as T;
    newClass.prototype = tmp.prototype;
    (newClass as any).getStores = (tmp as any).getStores;
    (newClass as any).calculateState = (tmp as any).calculateState;
    return newClass;
}
