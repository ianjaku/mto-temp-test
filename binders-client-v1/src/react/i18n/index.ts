import i18n from "../../i18n";
export { useTranslation, withTranslation } from "react-i18next";
export { type TFunction } from "../../i18n";
export default i18n;
