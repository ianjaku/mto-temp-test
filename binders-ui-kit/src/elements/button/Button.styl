@require "../../vars.styl";
@require "../../style/global.styl";

makeButtonDisabled()
    opacity: 0.2
    cursor: default

.button
    font-family: defaultFontName, sans-serif
    color: whiteColor
    background-color: baseColor
    padding: 10px 34px
    border-radius: 5px
    display: inline-flex
    align-items: center
    justify-content: center
    text-align: center
    transition: all 0.2s ease-in-out
    font-size: 13px
    font-weight: 500
    margin: 5px 0 5px 0
    position: relative
    text-decoration: none
    line-height: 1em
    border: 1px solid baseColor
    @media rwd.tabletAndAbove
        padding: 10px 40px
    &:not(.button--disabled)&:not(.button--unhoverable)
        cursor: pointer
        &:hover
            background-color: accentColor
            border-color: accentColor
    &:focus
        outline: none
    &--disabled
        makeButtonDisabled()
    &--secondary
        background-color: transparent
        border-color: baseColor
        color: baseColor
        &.branded
            border-color: var(--mainColor)
            color: var(--mainColor)
        &:not(.button--disabled)&:not(.button--unhoverable)
            &:hover
                border-color: baseColor
                background-color: baseColor
                color: white
                &.branded
                    border-color: var(--mainColor-hover)
                    background-color: var(--mainColor-hover)
        &^[0]--disabled
            makeButtonDisabled()
    &--CTA
        background-color: accentColor
        border-color: accentColor
        &.branded
            background-color: var(--mainColor)
            border-color: var(--mainColor)
        &:not(.button--disabled)&:not(.button--unhoverable)
            &:hover
                box-shadow: 0px 2px 3px 0px rgba(0,0,0,0.35)
                &.branded
                    background-color: var(--mainColor)
                    border-color: var(--mainColor)
        &^[0]--disabled
            makeButtonDisabled()
    &--borderless
        border: none
    &-round
        width: 24px;
        height: 24px;
        border-radius: 15px;
        background-color: accentColor;
        color: whiteColor;
        border: none;
        box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.2);
        cursor: pointer;
        display: inline-flex;
        align-items: center;
        justify-content: center;
        img
            box-shadow: none
        &:focus
            outline: none;

        .material-icons
            font-size: 20px !important;
            color: whiteColor !important;
    &-loader
        right: 10px
        top: 8px
        color: whiteColor !important
        &--secondary
            color: baseColor !important

    &-link
        position: absolute
        display: block
        top: 0
        bottom: 0
        right: 0
        left: 0

