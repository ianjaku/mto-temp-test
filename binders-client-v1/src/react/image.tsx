/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
import * as React from "react";
import { pick, replace } from "ramda";
import { isVideoURL } from "../clients/imageservice/v1/visuals";

export type Unit = "%" | "vw";

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
export function getFitCropStyleObject(fitBehaviour: string, aspectRatio: number, unit: Unit = "%") {
    const isLandscape = aspectRatio > 1;
    const addUnit = (amount: number) => `${amount}${unit}`;
    if (fitBehaviour === "fit") {
        if (isLandscape) {
            return {
                width: addUnit(100)
            };
        } else {
            const style = {
                height: addUnit(100),
                width: addUnit(Math.round(100 * aspectRatio))
            };
            return style;
        }
    } else {
        if (isLandscape) {
            return {
                left: addUnit(50),
                top: addUnit(50),
                height: addUnit(100),
                width: "auto",
                transform: `translate(-${addUnit(50)},-${addUnit(50)})`,
                position: "absolute"
            };
        } else {
            return {
                width: addUnit(100)
            };
        }
    }
}


export interface ImageProps {
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    image?: any;
    isOver?: boolean;
    dirtyDownSize?: boolean;
    className: string;
}

export interface ImageState {
    bgColor: string;
    width?: number;
    height?: number;
}

export default class Image extends React.Component<ImageProps, ImageState> {

    constructor(props) {
        super(props);
        this.state = {
            bgColor: props.image ? props.image.bgColor : "",
        };
    }

    handleImageLoaded(e) {
        this.setState({
            bgColor: this.props.image.bgColor,
            width: e.target.naturalWidth,
            height: e.target.naturalHeight
        });
    }

    dirtyDownsizeImageImg(img) {
        let url;
        if (img.url) {
            url = img.url;
        } else {
            url = img;
        }
        let replaceTarget = "/medium";
        if (isVideoURL(url)) {
            replaceTarget = "/video_screenshot";
        }
        return replace(/\/original/, replaceTarget, url);
    }

    getImageSrc() {
        if (this.props.dirtyDownSize) {
            return this.dirtyDownsizeImageImg(this.props.image);
        }
        if (this.props.image.buildRenderUrl) {
            return this.props.image.buildRenderUrl();
        }
        if (this.props.image.medium) {
            return this.props.image.medium;
        }
        return this.props.image;
    }

    getImageProps() {
        const { rotation }  = this.props.image;
        // eslint-disable-next-line @typescript-eslint/no-explicit-any
        const imageProps: any = {
            ...pick(["style", "alt"], this.props),
            className: this.props.isOver ? "hovering" : "",
            src: this.getImageSrc(),
            onLoad: this.handleImageLoaded.bind(this)
        };
        let rotationStyle = {};
        imageProps.style = Object.assign(imageProps.style ? imageProps.style : {}, this.getFitCropStyling());
        if(rotation) {
            const rotate = `rotate(${rotation}deg)`;
            rotationStyle = {transform:  (imageProps.style.transform) ? `${imageProps.style.transform} ${rotate}` : rotate};
            imageProps.style = {...imageProps.style, ...rotationStyle};
        }
        return imageProps;
    }

    getFitCropStyling() {
        return getFitCropStyleObject(this.props.image.fitBehaviour, this.getAspectRatio());
    }

    getAspectRatio() {
        return this.state.height ?
            this.state.width / this.state.height :
            1;
    }

    render() {
        const isTransparent = this.props.image.bgColor === "transparent";
        const wrapperStyle = { backgroundColor: isTransparent ? "inherit" : `#${this.props.image.bgColor}` } as React.CSSProperties;

        return (
            <div className={`${this.props.className ? this.props.className : ""} image-wrapper ${(isTransparent ? " transparent" : "")}`} style={wrapperStyle}>
                <div className="image-container">
                    <img {...this.getImageProps() } />
                </div>
            </div>
        );
    }

}
