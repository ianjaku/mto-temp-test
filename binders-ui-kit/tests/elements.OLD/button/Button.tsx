import * as React from "react";
import Button from "../../../src/elements/button";
import { create } from "react-test-renderer";
import { mount } from "enzyme";
import wrapWithTheme from "../../themeHelper";

const TITLE = "Button title";
let exampleValue = 100;
const onClick = () => exampleValue++;

const createBaseButton = (text) => wrapWithTheme(<Button text={text} onClick={onClick} />);
const createBaseButtonDisabled = (text) => wrapWithTheme(<Button text={text} onClick={onClick} isEnabled={false} />);
const createSecondaryButton = (text) => wrapWithTheme(<Button text={text} onClick={onClick} secondary={true} />);
const createSecondaryButtonDisabled = (text) => wrapWithTheme(<Button text={text} onClick={onClick} secondary={true} isEnabled={false} />);
const createCTAButton = (text) => wrapWithTheme(<Button text={text} onClick={onClick} CTA={true} />);
// eslint-disable-next-line @typescript-eslint/no-unused-vars
const createCTAButtonDisabled = (text) => wrapWithTheme(<Button text="CTA Button Disabld" onClick={onClick} CTA={true} isEnabled={false} />);

describe("Button", () => {

    beforeEach(() => {
        exampleValue = 100;
    });


    test("Base Button (snapshot)", () => {
        const modal = create(createBaseButton("Base Button"));
        const serialized = modal.toJSON();
        expect(serialized).toMatchSnapshot();
    });

    test("Base Button Disabled (snapshot)", () => {
        const modal = create(createBaseButtonDisabled("Base Button Disabled"));
        const serialized = modal.toJSON();
        expect(serialized).toMatchSnapshot();
    });

    test("Secondary Button (snapshot)", () => {
        const modal = create(createSecondaryButton("Secondary Button"));
        const serialized = modal.toJSON();
        expect(serialized).toMatchSnapshot();
    });

    test("Secondary Button disabled (snapshot)", () => {
        const modal = create(createSecondaryButtonDisabled("Secondary Button Disabled"));
        const serialized = modal.toJSON();
        expect(serialized).toMatchSnapshot();
    });

    test("CTA Button (snapshot)", () => {
        const modal = create(createCTAButton("CTA Button"));
        const serialized = modal.toJSON();
        expect(serialized).toMatchSnapshot();
    });

    test("CTA Button Disabled (snapshot)", () => {
        const modal = create(createCTAButtonDisabled("CTA Button Disabled"));
        const serialized = modal.toJSON();
        expect(serialized).toMatchSnapshot();
    });

    test("Button text (mount: enzyme+jsdom)", () => {
        const elementText = mount(createCTAButton(TITLE))
            .find(".button")
            .at(0)
            .text();
        expect(elementText).toEqual(TITLE);
    });

    test("Button disabled class (mount: enzyme+jsdom)", () => {
        const button = mount(createBaseButton(TITLE))
            .find(".button");
        expect(button.hasClass("button--disabled"));
    });

    test("Enabled Button onClick (shallow: enzyme+jsdom)", () => {
        // eslint-disable-next-line @typescript-eslint/no-unused-vars
        const button = mount(createSecondaryButton(TITLE))
            .find(".button")
            .simulate("click");
        expect(exampleValue).toEqual(101);
    });

    test("Disabled Button onClick (shallow: enzyme+jsdom)", () => {
        // eslint-disable-next-line @typescript-eslint/no-unused-vars
        const button = mount(createSecondaryButtonDisabled(TITLE))
            .find(".button")
            .simulate("click");
        expect(exampleValue).toEqual(100);
    });
});
