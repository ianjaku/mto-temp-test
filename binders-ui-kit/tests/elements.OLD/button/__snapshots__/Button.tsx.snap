// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Base Button (snapshot) 1`] = `
<div
  className="button"
  onBlur={undefined}
  onClick={[Function]}
  tabIndex={undefined}
>
  Base Button
  
  
</div>
`;

exports[`Button Base Button Disabled (snapshot) 1`] = `
<div
  className="button button--disabled"
  onBlur={undefined}
  onClick={[Function]}
  tabIndex={undefined}
>
  Base Button Disabled
  
  
</div>
`;

exports[`Button CTA Button (snapshot) 1`] = `
<div
  className="button button--CTA"
  onBlur={undefined}
  onClick={[Function]}
  tabIndex={undefined}
>
  CTA Button
  
  
</div>
`;

exports[`Button CTA Button Disabled (snapshot) 1`] = `
<div
  className="button button--disabled button--CTA"
  onBlur={undefined}
  onClick={[Function]}
  tabIndex={undefined}
>
  CTA Button Disabld
  
  
</div>
`;

exports[`Button Secondary Button (snapshot) 1`] = `
<div
  className="button button--secondary"
  onBlur={undefined}
  onClick={[Function]}
  tabIndex={undefined}
>
  Secondary Button
  
  
</div>
`;

exports[`Button Secondary Button disabled (snapshot) 1`] = `
<div
  className="button button--disabled button--secondary"
  onBlur={undefined}
  onClick={[Function]}
  tabIndex={undefined}
>
  Secondary Button Disabled
  
  
</div>
`;
